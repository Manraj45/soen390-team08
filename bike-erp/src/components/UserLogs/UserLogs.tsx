import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Axios from "axios";
import React, { useEffect, useState } from "react";
import useStyles from "../Inventory/inventoryStyles";

const UserLogs: React.FC = () => {
    const [logTable, setlogTable] = useState<any[]>([]);

  useEffect(() => {
    Axios.get("http://localhost:3001/userLogs/").then((response) => {
        setlogTable(response.data);
    });
  }, []);

  const classes = useStyles();

  return (
    <React.Fragment>
      <div id="userLogsPage" className={classes.background}>
        <br></br>
        <div className={classes.title}>User Logs</div>
        <br></br>
        <Table size="small" className={classes.tableStyle}>
          <TableHead className={classes.tableHead}>
            <TableRow className={classes.topRow}>
              <TableCell className={classes.emptyCell} />
              <TableCell>Time</TableCell>
              <TableCell>User</TableCell>
              <TableCell>Activity</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {logTable.map((row) => (
              <TableRow key={row.log_id}>
                <TableCell className={classes.tableHead} />
                <TableCell className={classes.innerTable}>
                  {row.time}
                </TableCell>
                <TableCell className={classes.innerTable}>
                  {row.user}
                </TableCell>
                <TableCell className={classes.innerTable}>
                  {row.activity}
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </div>
    </React.Fragment>
  );
};

export default UserLogs;