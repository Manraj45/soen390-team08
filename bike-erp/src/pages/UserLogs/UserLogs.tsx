// DEPENDENCIES
import Axios from "axios";
import React, { useEffect, useState } from "react";

// SERVICES
import { BACKEND_URL } from "../../core/utils/config";

// STYLING
import { Table, TableHead, TableBody, TableRow, TableCell } from "@material-ui/core";
import useStyles from "./UserLogsStyles";

/*
  The UserLogs page.
  Admins can view logs and audit trails of all users.
*/
const UserLogs: React.FC = () => {

  const styles = useStyles();
  const url = BACKEND_URL;

  const [logTable, setlogTable] = useState<any[]>([]);

  useEffect(() => {
    Axios.get(`${url}/userlogs/`).then((response) => {
      setlogTable(response.data);
    });
  }, [url]);

  return (
    <React.Fragment>
      <div id="userLogsPage" className={styles.background}>
        <br></br>
        <div className={styles.title}>User Logs</div>
        <br></br>
        <Table size="small" className={styles.tableStyle}>
          <TableHead className={styles.tableHead}>
            <TableRow className={styles.topRow}>
              <TableCell className={styles.emptyCell} />
              <TableCell>Time</TableCell>
              <TableCell>User</TableCell>
              <TableCell>Activity</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {logTable.map((row) => (
              <TableRow key={row.log_id}>
                <TableCell className={styles.tableHead} />
                <TableCell className={styles.innerTable}>
                  {row.timestamp}
                </TableCell>
                <TableCell className={styles.innerTable}>
                  {row.email}
                </TableCell>
                <TableCell className={styles.innerTable}>
                  {row.activity}
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </div>
    </React.Fragment>
  );
};

export default UserLogs;