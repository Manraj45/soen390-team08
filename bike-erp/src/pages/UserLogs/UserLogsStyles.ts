import { makeStyles } from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  title: {
    alignItems: "center",
    textAlign: "center",
    fontWeight: "bold",
    fontSize: "200%",
  },
  background: {
    background: "#DDDDDD",
    paddingLeft: "10px",
    paddingRight: "10px",
  },
  tableHead: {
    backgroundColor: "black",
    padding: "7px",
  },
  tableStyle: {
    width: "100%",
    marginLeft: "auto",
    marginRight: "auto",
  },
  innerTable: {
    background: "#FFFFFF",
  },
  topRow: {
    fontWeight: "bold",
    color: "white",
    background: "black",
    minWidth:"150px",
  },
  place: {
    maxHeight: "50px",
    width: "70%",
    marginLeft: "auto",
    marginRight: "auto",
    overflow: "auto",
    [theme.breakpoints.down("sm")]: {
      marginTop: "50px",
      minHeight: "500px",
    },
    [theme.breakpoints.up("md")]: {
      marginTop: "10px",
      minHeight: "650px",
    },
  },
}));

export default useStyles;