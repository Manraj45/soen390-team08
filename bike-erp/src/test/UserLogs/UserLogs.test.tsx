import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import * as React from "react";
import renderer from "react-test-renderer";
import UserLogs from "../../components/UserLogs/UserLogs";

Enzyme.configure({ adapter: new Adapter() });

describe("UserLogs", () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<UserLogs />);
  });

  it("userlogs page should appear", () => {
    console.log(wrapper);
    const data = wrapper.find("#userLogsPage");
    console.log(wrapper);
    expect(data.length).toBe(1);
  });

  it("userlogs render", () => {
    expect(wrapper.exists()).toBe(true);
  });

  it("matches the snapshot", () => {
    const tree = renderer.create(<UserLogs />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
