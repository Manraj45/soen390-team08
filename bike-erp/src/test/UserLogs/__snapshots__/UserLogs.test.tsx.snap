// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLogs matches the snapshot 1`] = `
<div
  className="makeStyles-background-2"
  id="userLogsPage"
>
  <br />
  <div
    className="makeStyles-title-1"
  >
    User Logs
  </div>
  <br />
  <div
    className="MuiPaper-root makeStyles-place-7 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      className="MuiTable-root makeStyles-tableStyle-4 MuiTable-stickyHeader"
      role={null}
    >
      <thead
        className="MuiTableHead-root makeStyles-tableHead-3"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topRow-6 MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
            scope="col"
          >
            Time
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topRow-6 MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
            scope="col"
          >
            User
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head makeStyles-topRow-6 MuiTableCell-sizeSmall MuiTableCell-stickyHeader"
            scope="col"
          >
            Activity
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      />
    </table>
  </div>
</div>
`;
